interpreter/cling/test/Prompt/ValuePrinter/SourceLocation.C

```diff
@@ -0,0 +1,16 @@
+//------------------------------------------------------------------------------
+// CLING - the C++ LLVM-based InterpreterG :)
+//
+// This file is dual-licensed: you can choose to license it under the University
+// of Illinois Open Source License or the GNU Lesser General Public License. See
+// LICENSE.TXT for details.
+
+//------------------------------------------------------------------------------
+
+// RUN: cat %s | %cling | FileCheck %s
+
+#if __cplusplus >= 202002L
+#include <source_location>
+std::source_location::current()
+// CHECK: (std::source_location) ROOT_prompt_0:2:__cling_Un1Qu30
```

Comment:
That won't work for cling where the "file" name does not start with "ROOT_".

Maybe add a `#line` directive, somehow?

Reply:
It also doesn't work in C++17 builds of Cling because `FileCheck` doesn't respect preprocessor `#if`...

Reply:
For adding `#line` directive, I can think of something like this:
```
#include <source_location>
__LINE__+1
// CHECK: (int) [[LINE]]
std::source_location::current()
// CHECK: (std::source_location) {{.*}}_prompt_{{[0-9]+}}:[[LINE]]:{{.*}}
```
but that looks very hacky. 

Maybe it's better to just have:
```
std::source_location::current()
// CHECK: (std::source_location) {{.*}}_prompt_{{[0-9]+}}:{{[0-9]+}}:{{.*}}
```
as we don't really need to check if current line and `__LINE__` are the same (it should be checked by the library).

Reply:
I was thinking of passing the filename in as part of the `#line` directive, but that won't work as I don't see a way to have `#line` and `std::source_location::current()` in the same transaction... Ah maybe:
```
auto getsrcloc() {
# line 42 "CHECK_SRCLOC"
  return std::source_location::current()
}
getsrcloc() // CHECK: (std::source_location) CHECK_SRCLOC:42
```
?